src/lib389/lib389/__init__.py

```diff
@@ -1730,6 +1730,16 @@ def with_systemd(self):
             return self.systemd_override
         return self.ds_paths.with_systemd
 
+    def with_systemd_running(self):
+        if not self.with_systemd():
+            return False
+        rc = subprocess.call(["systemctl", "is-system-running", "--quiet"])
+        # is-system-running can detect the 7 modes (initializing, starting,
+        # running, degraded, maintenance, stopping, offline) or "unknown".
+        # Exit code 0 is given for the "running" mode, which is the only one
+        # the systemd documentation describes as fully operational.
+        return rc == 0
```

Comment:
I have doubts about this check.
On my system I have one unit failed, and `systemctl is-system-running` reports `degraded`. 
So when I start the server using `dsctl localhost start`, it thinks that systemd is not running and starts the server outside of systemd:

```
sudo dsctl -v localhost start
DEBUG: The 389 Directory Server Administration Tool
DEBUG: Inspired by works of: ITS, The University of Adelaide
DEBUG: Called with: Namespace(verbose=True, instance='localhost', json=False, list=False, remove_all=False, func=<function instance_start at 0xffffa127f7e0>)
DEBUG: Allocate local instance <class 'lib389.DirSrv'> with None
DEBUG: Instance allocated
DEBUG: systemd status -> False
DEBUG: pid file /run/dirsrv/slapd-localhost.pid -> 74271
DEBUG: Pid of 74271 is not running for localhost
DEBUG: systemd status -> False
DEBUG: DEBUG: starting with ['/usr/sbin/ns-slapd', '-D', '/etc/dirsrv/slapd-localhost', '-i', '/run/dirsrv/slapd-localhost.pid']
INFO: Instance "localhost" has been started
```

```
systemctl status dirsrv@localhost | grep -i active
     Active: inactive (dead)
```

Reply:
The one other option that I considered was to check for the "offline" mode specifically. Or re-introduce the check for chroot environments with `systemd-detect-virt`.

Also, `is-system-running` has a `--wait` option that I think will prevent it from returning initializing, starting or stopping. That leaves the maintenance for which I'm not sure `systemctl start` will work or not.

Do you have any opinion about those choices? The rationale for checking for `offline` specifically is that we know for certain that `systemctl start` wont work there, and all other modes will keep the same behavior that they currently have.

Reply:
Yes, I think checking for `offline` mode specifically (as an indicator that systemd is not running as PID 1) is a better option. I checked kickstart chroot and container environments, where systemd is present but not running, they report `offline`.